.github/workflows/ci.yml

```diff
@@ -56,6 +82,35 @@ jobs:
         with:
           name: coverage
 
-      - uses: paambaati/codeclimate-action@v2.7.1 # using 2.7.1 for issue: https://github.com/paambaati/codeclimate-action/issues/316
+      - uses: paambaati/codeclimate-action@v2.7.3 # using 2.7.3 for issue: https://github.com/paambaati/codeclimate-action/issues/235
         env:
           CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
+        with:
+          debug: true
```

Comment:
right! Should have set that already considering how much I tried with this damn action. Like the idea of trying to figure it out in here. Even CodeClimate documentation suggests using this action, but I was already thinking of coming up with quick shell or js script just using that CodeClimate binary directly. But that action just working would be preferred of course

Reply:
I think I know what's wrong! Please update the action to most recent version but then we have to do a bit more.

Reply:
Here's what happens right now:

- all test runs will upload and overwrite the shared artifacts folder
- those coverage reports include file paths
- in our case, the test runs on Windows always finish last, so their uploaded artifact is the final one
- our `coverage` job then runs on Ubuntu and will download that coverage artifact folder, which then when trying to convert into CodeClimate format with the binary, will fail because the file system paths are different

That's why we see this:

<img width="624" alt="Screen Shot 2022-05-11 at 13 49 50" src="https://user-images.githubusercontent.com/90316/167853732-97a5e816-ad30-4a56-8996-237cbcfe49db.png">

As job runs on a unix file system it can't handle windows paths like `D:\a\market`

Solution: make artifact uploads system agnostic and then download the one generated with ubuntu

Reply:
Great! Hopefully this solves it, I'm kind of swimming in their github repo. I keep jumping from issue to issue, but without luck so far. At least now we'll have some more logs in debug, hopefully we see something useful.

Reply:
it looks like we now finally arrived at a proper error message not caused by our test setup:

<img width="960" alt="Screen Shot 2022-05-11 at 14 10 54" src="https://user-images.githubusercontent.com/90316/167857892-99672513-6b6c-4222-89df-d83b0b448513.png">

Indeed when browsing `coverage/` folder, the reports reference a `src/@types/subgraph/globalTypes.d.ts` but we actually don't have that. And also never access this from our source code...

Reply:
Yeah, but this is getting longer and longer. Maybe we should take this to it's actual [issue](https://github.com/oceanprotocol/market/issues/1409) to avoid making this longer by our tests.

Reply:
yup, just pushed last try. If that doesn't work I just roll back to get CI green without that coverage job for now

Reply:
trying to upload/download them as artifacts too didn't work so most likely the coverage job needs a full `npm ci` + `npm run codegen:apollo`